docs/frontend-architecture.md

```diff
@@ -115,3 +115,61 @@ If you want to add a new mock user, start by adding it to the `mockIds` array.
+
+## Communicate with the add-on
+
+There are two types of communication between the add-on and the website:
+sharing data, and notifications of events. And since this communication can
+happen in either direction, there are four situations to consider:
+
+### 1. The website shares data with the add-on
+
+This is mainly to tell the add-on which user is currently logged in, and data
+about them. (There are also plans to minimise this to only share the API key
+with the add-on, which can then fetch the rest of the data from the API
+directly, and even to have the add-on authenticate against Firefox Accounts
+directly and then being able to communicate with the API without even needing to
+interact with the website.)
+
+This is done via the `<firefox-private-relay-addon-data>` element, present in
+the user's dashboard, rendered by the `<AddonData>` component. The add-on
+looks for that element via its ID (`#profile-main`) and reads its attributes.
+
+Thus, if you want to share data with the add-on, you can add it as an attribute
+there. Then in the add-on, you can expand `get_profile_data.js` to read it and
+copy it over to the extension storage.
+
+### 2. The add-on shares data with the website
+
+The add-on mainly tells the website whether it is installed, and, in case the
+user has disabled server-side storage of mask labels, the labels it has stored
+locally. It does so by changing attributes on the
+`<firefox-private-relay-addon>` element, rendered in `_app.page.tsx`. React
+components looking to access that data can do so using the `useAddonData` hook.
+
+If you have more data to share with the website, you can add it in the add-on
+in `inject_addon_data.js`. To then make it available in the website, add the
+data you're injecting to the `AddonData` type in `/frontend/src/hooks/addon.ts`.
+Additionally, since attributes are always strings, you'll want to define how to
+convert that to the proper data structure (e.g. via `JSON.parse`) by adding a
+property to the `attributeParsers` object in that same file. The property name
+is the attribute you're writing, and the value is a function that takes a string
+(the attribute value) and returns the proper data structure.
+
+### 3. The website sends a notification to the add-on
+
+When e.g. the user performs an action on the website that the add-on should
+immediately act on, the website can send it a notification. This is used e.g. to
+tell the add-on to update the extension's storage when the user changes a mask
+label and has server-side storage disabled.
+
+To do so, React components can again use the `useAddonData` hook, whose return
+value also includes a `sendEvent` function. This then fires a `website` event on
+the `<firefox-private-relay-addon>` element, which is listened to by the add-on
+in `get_profile_data.js`.
+
+### 4. The add-on sends a notification to the website
+
+We don't actually do this. Instead, the add-on simply calls
```

Comment:
I know this is just documenting, but wanted to see if there was a use case for this?

Reply:
Yeah, I think we already discussed one since I created this PR: when the add-on generates an alias, it'd be nice if the website could re-fetch without having to reload the whole page.